core/src/main/scala/kafka/raft/KafkaNetworkChannel.scala

```diff
@@ -34,6 +34,7 @@ import scala.collection.mutable
 
 object KafkaNetworkChannel {
 
+  val nonRoutableAddress = new InetSocketAddress("0.0.0.0", 0)
```

Comment:
Let me suggest an alternative for the sake of argument. Currently, `RaftConfig.parseVoterConnections` return `Map<Integer, InetSocketAddress>`. This works for the case we're interested in, but there is a risk of our sentinel non-routable address leaking into unexpected cases (a common source of bugs in Kafkaland). Alternatively, what if we add something like this to `RaftConfig`:

```java
public class RaftConfig {
  ...

  public Map<Integer, AddressSpec> quorumVoterConnections();

  public static interface AddressSpec {
  }

  public static class InetAddressSpec implements AddressSpec {
    final InetSocketAddress address;
  }

  public static class UnknownAddressSpec implements AddressSpec {
  }
}
```

The advantage is that this lets the type checker help us ensure that we are checking for a sentinel.

Reply:
Yea, good catch. The AddressSpec makes sense.